fiat-roles/src/main/java/com/netflix/spinnaker/fiat/permissions/RedisPermissionsRepository.java

```diff
@@ -487,4 +515,60 @@ default R apply(T t) {
       super(source);
     }
   }
+
+  /**
+   * TimeoutContext allows specifying an expiration time for request processing.
+   *
+   * <p>If something exceeds the specified timeout duration, the request handler should stop doing
+   * work and just bail out with a timeout.
+   */
+  private static class TimeoutContext {
+    private final String name;
+    private final long timeoutAtMs;
```

Comment:
big time nit: `timeoutMs` seems more idiomatic than `timeoutAtMs`, not sure the `At` achieves much (though I know what you're going for).

Reply:
I would tend to see a value timeoutMs as something more like a duration than an instant

Reply:
Actually going to just change this to an Instant, I suspect this is just stupid over optimization!